clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -343,40 +411,125 @@ export class CopybookDownloadService {
     if (this.handleAsEndevorElement(documentUri)) {
       return !!(await this.e4eDownloader?.getE4EConfig(documentUri));
     }
-    if (!this.explorerApi) return false;
 
-    const copybooksLocation =
-      DownloadUtil.areCopybookDownloadConfigurationsPresent(
-        documentUri,
-        dialects,
-      );
+    const configs = await loadProcessorGroupCopybookPathsConfig(
+      { scopeUri: documentUri },
+      [],
+    );
 
-    if (!copybooksLocation) {
-      return false;
-    }
+    const procGroupZoweConfigs = configs
+      .filter(
+        (config): config is ZoweUssConfigModel | ZoweDatasetConfigModel =>
+          typeof config != "string" && (DATASET in config || USS in config),
+      )
+      .map((config) => ({
+        profile: config.profile,
+        dataset: DATASET in config ? config.dataset : undefined,
+        uss: USS in config ? config.uss : undefined,
+      }));
+
+    const endevorConfigs = configs.filter(
+      (config): config is EndevorConfigModel =>
+        typeof config != "string" && ENVIRONMENT in config,
+    );
 
     const profile = ProfileUtils.getProfileNameForCopybook(
       documentUri,
       this.explorerApi,
     );
-    const availableProfiles = ProfileUtils.getAvailableProfiles(
-      this.explorerApi,
-    );
-    if (!profile || !availableProfiles.includes(profile)) {
-      const message = profile
-        ? `${PROVIDE_PROFILE_MSG} Provided invalid profile name: ${profile}`
-        : `${PROVIDE_PROFILE_MSG}`;
-      this.processDownloadError(message);
+
+    if (endevorConfigs.length > 0 && !this.e4eApi) {
+      this.diagnosticsService?.showDiagnostics(vscode.Uri.parse(documentUri), [
+        {
+          range: new vscode.Range(
+            new vscode.Position(0, 0),
+            new vscode.Position(1, 0),
+          ),
+          message: "Explorer for Endevor is not installed",
+          severity: vscode.DiagnosticSeverity.Warning,
+        },
+      ]);
       return false;
     }
-    return (
-      !(await DownloadUtil.isProfileLocked(profile)) &&
-      !(await DownloadUtil.checkForInvalidCredProfile(
-        profile,
+    if (!this.explorerApi && procGroupZoweConfigs.length > 0) {
+      this.diagnosticsService?.showDiagnostics(vscode.Uri.parse(documentUri), [
+        {
+          range: new vscode.Range(
+            new vscode.Position(0, 0),
+            new vscode.Position(1, 0),
+          ),
+          message: "Zowe Explorer is not installed",
+          severity: vscode.DiagnosticSeverity.Warning,
+        },
+      ]);
+      return false;
+    }
+    if (this.explorerApi) {
+      const availableProfiles = ProfileUtils.getAvailableProfiles(
         this.explorerApi,
-        copybooksLocation,
-      ))
-    );
+      );
+
+      if (procGroupZoweConfigs && procGroupZoweConfigs.length > 0) {
+        const checks: boolean[] = [];
+        for (const zoweConfig of procGroupZoweConfigs) {
```

Comment:
two things here:
- profile will likely be the same for all the entries - so we can check if the profile is locked just for the distinct entries first.
- do we actually need to collect all the results into the checks array? I think it would be fine to just return as soon as the wrong answer is encountered.